challenges/backend/src/app/services/CarOnSaleClient/classes/Calculator.ts

```diff
@@ -0,0 +1,37 @@
+import { injectable } from "inversify";
+import { AuctionItem } from "../dtos/auction/AuctionItem";
+
+@injectable()
+export class Calculator {
+    calculatePercentage(currentHighestBidValue: number, minimumRequiredAsk: number | null):number {
+        let auctionProgress = 0;
+        if (minimumRequiredAsk === null) {
+            return 0;
+        }
+
+        auctionProgress = currentHighestBidValue / minimumRequiredAsk! * 100;
+        return auctionProgress;
+    }
```

Comment:
There is a problem here.
If you receive a valid minimumRequiredAsk as a numeric zero, it will lead you into a division by zero.

Reply:
Nice Catch :D